File: 프로그래머스/2/76502. 괄호 회전하기/괄호 회전하기.java
import java.util.*;
class Solution {
    public int solution(String s) {
        int answer = 0;
        for (int i = 0; i < s.length(); i++) {
            if (i != 0) {
                s = s.substring(1) + s.charAt(0);
            }
            answer++;
            Stack<Character> stack = new Stack<>();
            for (char ch : s.toCharArray()) {
                if (ch == '(' || ch == '[' || ch == '{') {
                    stack.push(ch);
                } else if (ch == ')' || ch == ']' || ch == '}') {
                    if (stack.isEmpty()) {
                        answer--;
                        break;
                    }
                    char top = stack.pop();
                    if ((ch == ')' && top != '(') || (ch == ']' && top != '[') || (ch == '}' && top != '{')) {
                        answer--;
                        stack.clear();
                        break;
                    }
                }
            }
            if (!stack.isEmpty()) {
                answer--;
            }
        }
        return answer;
    }
}

File: 프로그래머스/2/76502. 괄호 회전하기/괄호 회전하기.py
def solution(s):
    answer = 0
    for i in range(len(s)):
        if i != 0:
            s = s[1:] + s[0]
        answer += 1
        stack = []
        for ch in s:
            if ch in ['(', '[', '{']:
                stack.append(ch)
            elif ch in [')', ']', '}']:
                if not stack:
                    answer -= 1
                    break
                top = stack.pop()
                if (ch == ')' and top != '(') or (ch == ']' and top != '[') or (ch == '}' and top != '{'):
                    answer -= 1
                    stack.clear()
                    break
        if stack:
            answer -= 1
    return answer